### PR TITLE
rustc: 1.64.0 -> 1.65.0

### PR DESCRIPTION
**Pull request by tjni**
https://github.com/rust-lang/rust/releases/tag/1.65.0

I'm planning to run as many builds as I can to test this.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lblasc** [comment]
Upstream is using llvm15, we should wait for https://github.com/NixOS/nixpkgs/pull/194634

**Comment by tjni** [comment]
I'm temporarily pulling in unmerged LLVM update into this PR for testing.

**Comment by Fuuzetsu** [comment]
Might want to take #199300 at the same time as it causes full rust crate rebuilds too (though ofborg seems to not think so?)

**Comment by tjni** [comment]
If yours is ready, I'd put it on `staging` right away. I'm not yet sure when this will be ready, although I hope it's soon :)

**Comment by Fuuzetsu** [comment]
> If yours is ready, I'd put it on `staging` right away. I'm not yet sure when this will be ready, although I hope it's soon :)

I can at best send a PR as I don't have commits rights. Feel free to do it yourself if you'd like.

**Comment by tjni** [comment]
I don't have commit rights either, 🤞 your PR is reviewed soon.

**Comment by tjni** [comment]
1. Created a separate PR for https://github.com/NixOS/nixpkgs/pull/199794.
2. Wait for https://github.com/denoland/deno/issues/16540.

**Comment by oxalica** [comment]
@lblasc
> Upstream is using llvm15, we should wait for #194634

I found this in section "Compatibility Notes" of the release note,
> - Update the minimum external LLVM to 13

I guess LLVM 15 is not necessary required?

**Comment by lblasc** [comment]
hey @oxalica , off topic, thank you for all the work on rust in nix, on NixCon I heard about nocargo project which is awesome!

About the llvm compatibility, rust 1.65 should work on >=LLVM13, if I'm not mistaken rust upstream is not doing extensive testing on older LLVM versions, in past mismatched LLVM versions between nix and upstream caused some strange build and runtime issues in our monorepo which contains workspace with 80+ crates. That's the only argument I can provide, but sure LLVM14 should work just fine, maybe we can update afterwards.

**Comment by kevincox** [comment]
I was trying to build https://github.com/NixOS/nixpkgs/commit/763c18eb1e6d4160c47d8cf3c6d09ab02156f64d and it seemed mostly successful. Unfortunately it didn't finish as a few packages appear to be hung:

> 4/11111/11124 built (179 failed)

However trying to build 137017341b809b8a0f01576849331cbafe756283 I am getting an eval error.

```
% nixpkgs-review pr --no-shell --post-result -c commit $(c) --build-args '-j4 --cores 32 --max-silent-time 3600'
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs staging:refs/nixpkgs-review/0 pull/199664/head:refs/nixpkgs-review/1
$ git worktree add /home/kevincox/.cache/nixpkgs-review/pr-199664-11/nixpkgs 137017341b809b8a0f01576849331cbafe756283
Preparing worktree (detached HEAD 137017341b8)
Updating files: 100% (32616/32616), done.
HEAD is now at 137017341b8 rustc: 1.64.0 -> 1.65.0
error: Function called without required argument "libusb" at /home/kevincox/.cache/nixpkgs-review/pr-199664-11/nixpkgs/pkgs/development/tools/rust/humility/default.nix:4, did you mean "librsb", "libusb1" or "ibus"?
(use '--show-trace' to show detailed location information)
nix --experimental-features nix-command --system x86_64-linux eval --json --impure --expr (import /nix/store/nci1s3y693yifx3kqg0cd81p8alybnqk-nixpkgs-review-2.7.0/lib/python3.10/site-packages/nixpkgs_review/nix/evalAttrs.nix { allowAliases = false; attr-json = /run/user/1000/tmpht_x_zp3; }) failed to run, /run/user/1000/tmpht_x_zp3 was stored inspection
https://github.com/NixOS/nixpkgs/pull/199664 failed to build
```

**Comment by tjni** [comment]
1. Sorry, I messed up rebasing somehow originally. Fixed now.
2. @kevincox Thank you for testing this! The error you're seeing will be fixed by https://github.com/NixOS/nixpkgs/pull/199794

File: pkgs/development/web/deno/default.nix
Comment (06kellyjac) [inline pkgs/development/web/deno/default.nix:29]: If you're updating these to hash can `fetchFromGitHub` also use `hash`
And can you remove this space

**Comment by kevincox** [comment]
Something seems up with the latest version, I see lots of breakage: [nixpkgs-review.log](https://github.com/NixOS/nixpkgs/files/9995183/nixpkgs-review.log)


649 packages marked as broken and skipped:
aqemu atom-beta autotrace binance blockbench-electron breitbandmessung cinnamon.xapps clockify couchdb3 cwltool electron_10 electron_11 electron_12 electron_13 electron_14 electron_15 electron_16 electron_17 electron_9 elementary-planner emacsPackages.cedille emacsPackages.eterm-256color emacsPackages.ghc-mod emacsPackages.instapaper emacsPackages.magit-stgit emacsPackages.melancholy-theme emacsPackages.nemerle emacsPackages.rcirc-menu emacsPackages.sql-presto etcher gimpPlugins.exposureBlend gnome.gnome-todo gnome2.gnome-icon-theme gnome2.gnome_icon_theme gnome2.gnomeicontheme gnome2.gvfs gnome2.python_rsvg gscrabble hyper-haskell intensity-normalization invoice2data itch jabref jfbview jira-cli jitsi-meet-electron jucipp k2pdfopt kexi lexicon libsForQt512.akonadi libsForQt512.akonadi-calendar libsForQt512.akonadi-calendar-tools libsForQt512.akonadi-contacts libsForQt512.akonadi-import-wizard libsForQt512.akonadi-mime libsForQt512.akonadi-notes libsForQt512.akonadi-search libsForQt512.akonadiconsole libsForQt512.akregator libsForQt512.alkimia libsForQt512.angelfish libsForQt512.applet-window-buttons libsForQt512.ark libsForQt512.baloo libsForQt512.baloo-widgets libsForQt512.bismuth libsForQt512.bluedevil libsForQt512.bomber libsForQt512.bovo libsForQt512.breeze-gtk libsForQt512.breeze-qt5 libsForQt512.buho libsForQt512.calendarsupport libsForQt512.clip libsForQt512.communicator libsForQt512.discover libsForQt512.dolphin libsForQt512.dolphin-plugins libsForQt512.dragon libsForQt512.elisa libsForQt512.eventviews libsForQt512.falkon libsForQt512.ffmpegthumbs libsForQt512.filelight libsForQt512.frameworkintegration libsForQt512.granatier libsForQt512.grantleetheme libsForQt512.gwenview libsForQt512.incidenceeditor libsForQt512.index libsForQt512.k3b libsForQt512.kaccounts-integration libsForQt512.kaccounts-providers libsForQt512.kactivities libsForQt512.kactivities-stats libsForQt512.kactivitymanagerd libsForQt512.kaddressbook libsForQt512.kalarm libsForQt512.kalendar libsForQt512.kalzium libsForQt512.kamoso libsForQt512.kapman libsForQt512.kapptemplate libsForQt512.kasts libsForQt512.kate libsForQt512.katomic libsForQt512.kblackbox libsForQt512.kblocks libsForQt512.kbounce libsForQt512.kbreakout libsForQt512.kcachegrind libsForQt512.kcalc libsForQt512.kcalutils libsForQt512.kclock libsForQt512.kcmutils libsForQt512.kdav libsForQt512.kde-cli-tools libsForQt512.kde-gtk-config libsForQt512.kdeclarative libsForQt512.kdeconnect-kde libsForQt512.kded libsForQt512.kdegraphics-mobipocket libsForQt512.kdegraphics-thumbnailers libsForQt512.kdelibs4support libsForQt512.kdenetwork-filesharing libsForQt512.kdenlive libsForQt512.kdepim-addons libsForQt512.kdepim-runtime libsForQt512.kdeplasma-addons libsForQt512.kdesignerplugin libsForQt512.kdev-php libsForQt512.kdev-python libsForQt512.kdevelop libsForQt512.kdevelop-unwrapped libsForQt512.kdewebkit libsForQt512.kdf libsForQt512.kdialog libsForQt512.kdiamond libsForQt512.keditbookmarks libsForQt512.kfind libsForQt512.kget libsForQt512.kgpg libsForQt512.khelpcenter libsForQt512.khotkeys libsForQt512.khtml libsForQt512.kidentitymanagement libsForQt512.kig libsForQt512.kigo libsForQt512.killbots libsForQt512.kimageformats libsForQt512.kimap libsForQt512.kinfocenter libsForQt512.kinit libsForQt512.kio libsForQt512.kio-extras libsForQt512.kio-gdrive libsForQt512.kipi-plugins libsForQt512.kldap libsForQt512.kleopatra libsForQt512.klettres libsForQt512.klines libsForQt512.kmag libsForQt512.kmahjongg libsForQt512.kmail libsForQt512.kmail-account-wizard libsForQt512.kmailtransport libsForQt512.kmediaplayer libsForQt512.kmenuedit libsForQt512.kmines libsForQt512.kmix libsForQt512.kmplot libsForQt512.knavalbattle libsForQt512.knetwalk libsForQt512.knewstuff libsForQt512.knights libsForQt512.knotes libsForQt512.knotifyconfig libsForQt512.koko libsForQt512.kolf libsForQt512.kollision libsForQt512.kolourpaint libsForQt512.kompare libsForQt512.konqueror libsForQt512.konquest libsForQt512.konsole libsForQt512.kontact libsForQt512.kontactinterface libsForQt512.konversation libsForQt512.korganizer libsForQt512.kparts libsForQt512.kpat libsForQt512.kpimtextedit libsForQt512.kpmcore libsForQt512.kpurpose libsForQt512.krdc libsForQt512.kreport libsForQt512.kreversi libsForQt512.krohnkite libsForQt512.kross libsForQt512.krunner libsForQt512.krunner-symbols libsForQt512.kscreen libsForQt512.kscreenlocker libsForQt512.kshisen libsForQt512.ksmtp libsForQt512.kspaceduel libsForQt512.ksquares libsForQt512.ksudoku libsForQt512.ksystemlog libsForQt512.ksystemstats libsForQt512.kteatime libsForQt512.ktexteditor libsForQt512.ktimer libsForQt512.ktnef libsForQt512.ktorrent libsForQt512.ktouch libsForQt512.kturtle libsForQt512.kwalletmanager libsForQt512.kwave libsForQt512.kweather libsForQt512.kwin libsForQt512.kwin-dynamic-workspaces libsForQt512.kwin-tiling libsForQt512.kxmlrpcclient libsForQt512.kzones libsForQt512.libgravatar libsForQt512.libkcddb libsForQt512.libkdegames libsForQt512.libkdepim libsForQt512.libkgapi libsForQt512.libkleo libsForQt512.libkomparediff2 libsForQt512.libksieve libsForQt512.libksysguard libsForQt512.libktorrent libsForQt512.mailcommon libsForQt512.mailimporter libsForQt512.marble libsForQt512.mauikit-accounts libsForQt512.mauikit-filebrowsing libsForQt512.mauikit-imagetools libsForQt512.mauikit-texteditor libsForQt512.mbox-importer libsForQt512.messagelib libsForQt512.milou libsForQt512.neochat libsForQt512.nota libsForQt512.okular libsForQt512.oxygen libsForQt512.parachute libsForQt512.phonon-backend-gstreamer libsForQt512.phonon-backend-vlc libsForQt512.picmi libsForQt512.pim-data-exporter libsForQt512.pim-sieve-editor libsForQt512.pimcommon libsForQt512.pix libsForQt512.plasma-applet-caffeine-plus libsForQt512.plasma-applet-virtual-desktop-bar libsForQt512.plasma-bigscreen libsForQt512.plasma-browser-integration libsForQt512.plasma-desktop libsForQt512.plasma-dialer libsForQt512.plasma-disks libsForQt512.plasma-framework libsForQt512.plasma-integration libsForQt512.plasma-mobile libsForQt512.plasma-nano libsForQt512.plasma-nm libsForQt512.plasma-pa libsForQt512.plasma-remotecontrollers libsForQt512.plasma-sdk libsForQt512.plasma-settings libsForQt512.plasma-systemmonitor libsForQt512.plasma-thunderbolt libsForQt512.plasma-vault libsForQt512.plasma-workspace libsForQt512.plasmatube libsForQt512.powerdevil libsForQt512.print-manager libsForQt512.qtcurve libsForQt512.rocs libsForQt512.sddm-kcm libsForQt512.shelf libsForQt512.skanlite libsForQt512.skanpage libsForQt512.soundkonverter libsForQt512.spacebar libsForQt512.spectacle libsForQt512.systemsettings libsForQt512.umbrello libsForQt512.vvave libsForQt512.xdg-desktop-portal-kde libsForQt512.yakuake libsForQt512.zanshin libsForQt514.akonadi libsForQt514.akonadi-calendar libsForQt514.akonadi-calendar-tools libsForQt514.akonadi-contacts libsForQt514.akonadi-import-wizard libsForQt514.akonadi-mime libsForQt514.akonadi-notes libsForQt514.akonadi-search libsForQt514.akonadiconsole libsForQt514.akregator libsForQt514.alkimia libsForQt514.angelfish libsForQt514.applet-window-buttons libsForQt514.ark libsForQt514.baloo libsForQt514.baloo-widgets libsForQt514.bismuth libsForQt514.bluedevil libsForQt514.bomber libsForQt514.bovo libsForQt514.breeze-gtk libsForQt514.breeze-qt5 libsForQt514.buho libsForQt514.calendarsupport libsForQt514.clip libsForQt514.communicator libsForQt514.discover libsForQt514.dolphin libsForQt514.dolphin-plugins libsForQt514.dragon libsForQt514.elisa libsForQt514.eventviews libsForQt514.falkon libsForQt514.ffmpegthumbs libsForQt514.filelight libsForQt514.frameworkintegration libsForQt514.granatier libsForQt514.grantleetheme libsForQt514.gwenview libsForQt514.incidenceeditor libsForQt514.index libsForQt514.k3b libsForQt514.kaccounts-integration libsForQt514.kaccounts-providers libsForQt514.kactivities libsForQt514.kactivities-stats libsForQt514.kactivitymanagerd libsForQt514.kaddressbook libsForQt514.kalarm libsForQt514.kalendar libsForQt514.kalzium libsForQt514.kamoso libsForQt514.kapman libsForQt514.kapptemplate libsForQt514.kasts libsForQt514.kate libsForQt514.katomic libsForQt514.kblackbox libsForQt514.kblocks libsForQt514.kbounce libsForQt514.kbreakout libsForQt514.kcachegrind libsForQt514.kcalc libsForQt514.kcalutils libsForQt514.kclock libsForQt514.kcmutils libsForQt514.kdav libsForQt514.kde-cli-tools libsForQt514.kde-gtk-config libsForQt514.kdeclarative libsForQt514.kdeconnect-kde libsForQt514.kded libsForQt514.kdegraphics-mobipocket libsForQt514.kdegraphics-thumbnailers libsForQt514.kdelibs4support libsForQt514.kdenetwork-filesharing libsForQt514.kdenlive libsForQt514.kdepim-addons libsForQt514.kdepim-runtime libsForQt514.kdeplasma-addons libsForQt514.kdesignerplugin libsForQt514.kdev-php libsForQt514.kdev-python libsForQt514.kdevelop libsForQt514.kdevelop-unwrapped libsForQt514.kdewebkit libsForQt514.kdf libsForQt514.kdialog libsForQt514.kdiamond libsForQt514.keditbookmarks libsForQt514.kfind libsForQt514.kget libsForQt514.kgpg libsForQt514.khelpcenter libsForQt514.khotkeys libsForQt514.khtml libsForQt514.kidentitymanagement libsForQt514.kig libsForQt514.kigo libsForQt514.killbots libsForQt514.kimageformats libsForQt514.kimap libsForQt514.kinfocenter libsForQt514.kinit libsForQt514.kio libsForQt514.kio-extras libsForQt514.kio-gdrive libsForQt514.kipi-plugins libsForQt514.kldap libsForQt514.kleopatra libsForQt514.klettres libsForQt514.klines libsForQt514.kmag libsForQt514.kmahjongg libsForQt514.kmail libsForQt514.kmail-account-wizard libsForQt514.kmailtransport libsForQt514.kmediaplayer libsForQt514.kmenuedit libsForQt514.kmines libsForQt514.kmix libsForQt514.kmplot libsForQt514.knavalbattle libsForQt514.knetwalk libsForQt514.knewstuff libsForQt514.knights libsForQt514.knotes libsForQt514.knotifyconfig libsForQt514.koko libsForQt514.kolf libsForQt514.kollision libsForQt514.kolourpaint libsForQt514.kompare libsForQt514.konqueror libsForQt514.konquest libsForQt514.konsole libsForQt514.kontact libsForQt514.kontactinterface libsForQt514.konversation libsForQt514.korganizer libsForQt514.kparts libsForQt514.kpat libsForQt514.kpimtextedit libsForQt514.kpmcore libsForQt514.kpurpose libsForQt514.krdc libsForQt514.kreport libsForQt514.kreversi libsForQt514.krohnkite libsForQt514.kross libsForQt514.krunner libsForQt514.krunner-symbols libsForQt514.kscreen libsForQt514.kscreenlocker libsForQt514.kshisen libsForQt514.ksmtp libsForQt514.kspaceduel libsForQt514.ksquares libsForQt514.ksudoku libsForQt514.ksystemlog libsForQt514.ksystemstats libsForQt514.kteatime libsForQt514.ktexteditor libsForQt514.ktimer libsForQt514.ktnef libsForQt514.ktorrent libsForQt514.ktouch libsForQt514.kturtle libsForQt514.kwalletmanager libsForQt514.kwave libsForQt514.kweather libsForQt514.kwin libsForQt514.kwin-dynamic-workspaces libsForQt514.kwin-tiling libsForQt514.kxmlrpcclient libsForQt514.kzones libsForQt514.libgravatar libsForQt514.libkcddb libsForQt514.libkdegames libsForQt514.libkdepim libsForQt514.libkgapi libsForQt514.libkleo libsForQt514.libkomparediff2 libsForQt514.libksieve libsForQt514.libksysguard libsForQt514.libktorrent libsForQt514.mailcommon libsForQt514.mailimporter libsForQt514.marble libsForQt514.mauikit-accounts libsForQt514.mauikit-filebrowsing libsForQt514.mauikit-imagetools libsForQt514.mauikit-texteditor libsForQt514.mbox-importer libsForQt514.messagelib libsForQt514.milou libsForQt514.neochat libsForQt514.nota libsForQt514.okular libsForQt514.oxygen libsForQt514.parachute libsForQt514.phonon-backend-gstreamer libsForQt514.phonon-backend-vlc libsForQt514.picmi libsForQt514.pim-data-exporter libsForQt514.pim-sieve-editor libsForQt514.pimcommon libsForQt514.pix libsForQt514.plasma-applet-caffeine-plus libsForQt514.plasma-applet-virtual-desktop-bar libsForQt514.plasma-bigscreen libsForQt514.plasma-browser-integration libsForQt514.plasma-desktop libsForQt514.plasma-dialer libsForQt514.plasma-disks libsForQt514.plasma-framework libsForQt514.plasma-integration libsForQt514.plasma-mobile libsForQt514.plasma-nano libsForQt514.plasma-nm libsForQt514.plasma-pa libsForQt514.plasma-remotecontrollers libsForQt514.plasma-sdk libsForQt514.plasma-settings libsForQt514.plasma-systemmonitor libsForQt514.plasma-thunderbolt libsForQt514.plasma-vault libsForQt514.plasma-workspace libsForQt514.plasmatube libsForQt514.powerdevil libsForQt514.print-manager libsForQt514.qtcurve libsForQt514.rocs libsForQt514.sddm-kcm libsForQt514.shelf libsForQt514.skanlite libsForQt514.skanpage libsForQt514.soundkonverter libsForQt514.spacebar libsForQt514.spectacle libsForQt514.systemsettings libsForQt514.umbrello libsForQt514.vvave libsForQt514.xdg-desktop-portal-kde libsForQt514.yakuake libsForQt514.zanshin linuxKernel.packages.hardkernel_4_14.virtualbox linuxPackages_hardkernel_latest.virtualbox lispPackages_new.sbclPackages.nyxt_slash_benchmark lispPackages_new.sbclPackages.nyxt_slash_documentation lispPackages_new.sbclPackages.nyxt_slash_gi-gtk lispPackages_new.sbclPackages.nyxt_slash_gi-gtk-application lispPackages_new.sbclPackages.nyxt_slash_gtk lispPackages_new.sbclPackages.nyxt_slash_gtk-application lispPackages_new.sbclPackages.nyxt_slash_tests lispPackages_new.sbclPackages.nyxt_slash_version lua52Packages.nfd lua53Packages.nfd luaPackages.nfd mailnag mailnagWithPlugins markmind micropad morgen nagstamon nasc obinskit open-stage-control pantheon.elementary-tasks pantheon.evince pantheon.file-roller pocket-casts pomotroid pypi2nix python310Packages.accupy python310Packages.beaker python310Packages.caffe python310Packages.caffeWithCuda python310Packages.descartes python310Packages.graspologic python310Packages.hug python310Packages.imgaug python310Packages.intensity-normalization python310Packages.jupyter-book python310Packages.mapbox python310Packages.mask-rcnn python310Packages.mlxtend python310Packages.optuna python310Packages.perfplot python310Packages.polars python310Packages.pygbm python310Packages.pymatgen-lammps python310Packages.pyramid_beaker python310Packages.pytest-ansible python310Packages.pytest-sanic python310Packages.rl-coach python310Packages.robomachine python310Packages.schema-salad python310Packages.sklearn-deap python310Packages.skorch python310Packages.solo-python python310Packages.tiledb python310Packages.transip python310Packages.tvdb_api python310Packages.xgboost python39Packages.accupy python39Packages.beaker python39Packages.descartes python39Packages.graspologic python39Packages.hug python39Packages.imgaug python39Packages.intensity-normalization python39Packages.jupyter-book python39Packages.mapbox python39Packages.mask-rcnn python39Packages.mlxtend python39Packages.optuna python39Packages.perfplot python39Packages.polars python39Packages.pygbm python39Packages.pymatgen-lammps python39Packages.pyramid_beaker python39Packages.pytest-ansible python39Packages.pytest-sanic python39Packages.rl-coach python39Packages.robomachine python39Packages.schema-salad python39Packages.sklearn-deap python39Packages.skorch python39Packages.solo-python python39Packages.tiledb python39Packages.transip python39Packages.tvdb_api scribus_1_4 sparrow super-productivity tdlib-purple teleprompter thrift-0_10 todoist-electron truvari tvnamer vimPlugins.coc-imselect xfce.thunar-bare xfce.xfce4-embed-plugin xfce.xfce4-hardware-monitor-plugin xfce.xfce4-namebar-plugin xmonad_log_applet xmonad_log_applet_xfce xprite-editor yandex-browser zerobin zeronet

4 packages blacklisted:
appimage-run-tests nixos-install-tools tests.nixos-functions.nixos-test tests.writers

14581 packages failed to build: [see attachment]

29 packages built:
binaryen clang-analyzer clang-tools clang_15 cvise dtools ldc lit lld_15 lldb_10 lldb_11 lldb_9 llvmPackages_14.openmp llvmPackages_15.bintools llvmPackages_15.bintools-unwrapped llvmPackages_15.bintoolsNoLibc llvmPackages_15.clang-manpages llvmPackages_15.libclang llvmPackages_15.clangNoCompilerRt llvmPackages_15.clangNoCompilerRtWithLibc llvmPackages_15.clangNoLibc llvmPackages_15.clangNoLibcxx llvmPackages_15.compiler-rt llvm_15 llvmPackages_15.llvm-manpages llvmPackages_15.openmp llvmPackages_15.stdenv rund vscode-extensions.ms-vscode.cpptools

File: pkgs/development/web/deno/default.nix
Comment (tjni) [reply pkgs/development/web/deno/default.nix:29]: Absolutely, done!

**Comment by tjni** [comment]
I hope that error is resolved for you soon. It sounds painful from reading the issue.

**Comment by kevincox** [comment]
It seems to be triggering frequently on that version of this PR for some reason so I'm having trouble testing. Is there anything holding this back though? May make sense to dump it into stshung if we don't have any known issues.

**Comment by tjni** [comment]
The main question I would need an owner to resolve is whether we want to track upstream's LLVM exactly. This is requested in a comment that, right now, is in `1_64.nix`, which was originally committed in https://github.com/NixOS/nixpkgs/pull/76100 by @edolstra. @cstrahan @globin and @Havvy are maintainers of rustc whose opinions I think should matter.

If so, we need to wait for https://github.com/NixOS/nixpkgs/pull/194634 to land. If not, I can update this PR to do without that and create one to follow up when that's available. Without any owner opinion, I plan to wait for the LLVM update to land.

I could not run nixpkgs-review, because it required too much disk space and time. However, I did build a large sample of the applications listed on https://github.com/rust-unofficial/awesome-rust in my aarch64-darwin machine and on an x86_64-Linux machine from GitHub Actions.

If it aids you in testing, on my Cachix cache **tjni**, I have LLVM 15 built from a previous iteration of https://github.com/NixOS/nixpkgs/pull/194634. I haven't yet, but I'm going to spin up a GitHub Actions runner to build firefox and thunderbird.

**Comment by oxalica** [comment]
@lblasc 
> About the llvm compatibility, rust 1.65 should work on >=LLVM13, if I'm not mistaken rust upstream is not doing extensive testing on older LLVM versions, in past mismatched LLVM versions between nix and upstream caused some strange build and runtime issues in our monorepo which contains workspace with 80+ crates. That's the only argument I can provide, but sure LLVM14 should work just fine, maybe we can update afterwards.

I'd prefer using the current LLVM14 for now if #194634 cannot be merged in the near future. While rust 1.65.0, which is a big update, will be required for more and more crates targeting latest stable compiler, this PR would block all of them.

**Comment by kevincox** [comment]
I've been building the latest commit and it doesn't appear that there are any major breakages. I think it makes sense to move this into staging.

**Comment by tjni** [comment]
Thanks, everyone. I've updated the commit to use LLVM 14 and extracted the the deno fix into https://github.com/NixOS/nixpkgs/pull/201122.

**Review by kevincox — APPROVED** [review]

**Comment by Mic92** [comment]
x86_64-linux: firefox built
aarch64-macos: fd built
aarch64-linux: still building.

**Pull request merged**
